src/NUnitFramework/framework/Attributes/OSPlatformTranslator.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+
+namespace NUnit.Framework
+{
+    /// <summary>
+    /// Marks an assembly, test fixture or test method as applying to a specific platform.
+    /// </summary>
+    public static class OSPlatformTranslator
+    {
+        /// <summary>
+        /// Converts a .NET 5+ SupportedOSPlatformAttribute into an NUnit PlatformAttribute
+        /// </summary>
+        /// <param name="possibleSupportedOSPlatformAttribute"></param>
+        /// <returns></returns>
+        public static object Translate(object possibleSupportedOSPlatformAttribute)
```

Comment:
Personally I would prefer something like that:
```csharp
public static bool TryTranslate(object possibleSupportedOSPlatformAttribute, out PlatformAttribute platformAttribute);
```

That would make it more obvious what this method returns.

And then use it smth like that:
```csharp
OSPlatformTranslator.TryTranslate(attribute, out var platformAttribute) ? platformAttribute : attribute
```

(Your option is more concise though, so I don't insist to hard)

Reply:
Because of #3941 I have to changes this, combining multiple `OSPlatformAttribute` into a single `PlatformAttribute`

Reply:
Thanks for explaining this, I was going to ask why the need to handle `PlatformAttribute` in the translator. Are you saying that this PR will also fix https://github.com/nunit/nunit/issues/3941 ?

Reply:
@stevenaw This PR will as a side effect combine multiple Platform attributes into one, thereby 'fixing' #3941. It depends if the owners want that or change the AllowMultiple flag.
The main reason I combine it here is for code that currently use both `SupportOSPlatform` and `Platform`. The first for the .NET compiler, the 2nd for nunit.

Reply:
@manfred-brands Sorry it's taken me a few days to get back to you. That makes a lot of sense, thanks!

Reply:
I think it is fix to fix #3941 and keep the AllowMultiple as it is.